### ISSUE TITLE
Custom autocomplete/suggest

### ISSUE THREAD

**Issue by Lonelind** [erikras/redux-form#106]
Hi! I am facing the problem, I can't resolve.

I have many forms in my SPA and they have multiple custom selects with suggest logic inside (I can't change this — I'm just rewriting the whole project to React+Redux based stack). And there are different types of them:

1. They could have a static lists of items
2. They could have lists of items, that are fetching from the server once a component has been mounted
3. They could fetch actual list of items every time you change an input.

So, the question is: how to implement a dynamic list of values corresponding to your redux-form to match representative value and actual data that lies beneath?

It looks like this:
```jsx
return (
    <div>
        <input 
            type="text"
            value={this.props.displayValue}
            readOnly={this.props.type === 'select'}
            onBlur={(evt) => this.toggleSelectable(false, evt)}
            onFocus={(evt) => this.toggleSelectable(true, evt)}
        />
        {!this.state.empty && <span onClick={(evt) => this.clearInput(evt)}>×</span>}
        {this.state.opened && (
            <div className="items">
                <div className="list" onWheel={(evt) => this.scrollItems(evt)}>
                    {(this.props.items.map((item) => <div key={item.value} onClick={(evt) => this.selectItem(item, evt)}>{item.displayValue}</div> )}
                </div>
            </div>
        )}
    </div>
);
```

So, I have an ```<input>``` to catch user search, that could be readonly prior to make a regular select with custom design of this component. And I have a minimal state shape that allows us to know if my component opened, empty or disabled (i pass a condition to calculate it). 

All I need to know is where to store my items and how to pass a selected value to your form implementation when I catch selection action.

**Comment by Lonelind** [comment]
Thanks for your response. I did it myself.

**Comment by erikras** [comment]
I apologize. I totally missed this issue. As you've probably figured out, passing the selected value could be done with either `handleBlur` or `handleChange`. I have been using `react-select` to do something very similar. It keeps all the state that you're having to deal with, and then just passes the selection `onChange`.

**Comment by Lonelind** [comment]
I've made some selectors with ```react-select``` and then used my redux store to collect items in ```autocomplete``` field, where i store both current value (that displays inside an ```<input>```) and current selection if I ever clicked any item. So, in advance, I guess I should pass form name to prevent naming conflicts between fields inside two or more forms. I'm new to redux, and store refactoring, that my colleague advices me to do, helped me much more than anything I tried to do with my component. Maybe sometimes I will split app logic from this component and make a package. Thanks and sorry
